cms/static/js/views/pages/container.js

```diff
@@ -131,10 +133,40 @@ function($, _, Backbone, gettext, BasePage,
 
             if (this.options.isIframeEmbed) {
                 window.addEventListener('message', (event) => {
-                  if (event.data && event.data.type === 'refreshXBlock') {
-                      this.render();
-                  }
-               });
+                    const { data } = event;
+
+                    if (!data) return;
+
+                    const xblockElement = this.findXBlockElement(this.targetXBlock);
```

Comment:
Would it be better to include the XBlock ID in the event payload rather than keeping track of it via `targetXBlock` ? I see `data.payload.locator` below - how do we know that's the same as `targetXBlock` ?

Reply:
Agree, good idea
Added